src/components/collapsible/Collapsible.js

```diff
@@ -50,30 +58,45 @@ class Collapsible extends React.Component<Props, State> {
 
     constructor(props: Props) {
         super(props);
+        this.transitionElement = React.createRef();
         this.state = {
             isOpen: props.isOpen,
+            height: 'auto',
         };
     }
 
+    componentDidMount() {
+        const { current } = this.transitionElement;
+        if (current) {
+            this.setState(prevState => ({
+                height: prevState.isOpen ? `${current.scrollHeight}px` : '0',
+            }));
+        }
+    }
+
     toggleVisibility = () => {
         const { onOpen, onClose } = this.props;
-        this.setState(
-            prevState => ({
-                isOpen: !prevState.isOpen,
-            }),
-            () => {
-                const { isOpen } = this.state;
-                if (isOpen && onOpen) {
-                    onOpen(this);
-                } else if (!isOpen && onClose) {
-                    onClose(this);
-                }
-            },
-        );
+        const { current } = this.transitionElement;
+        if (current) {
```

Comment:
Minor stylistic comment:
```
if (!current) {
    return;
}
```
In this way majority of your function is less indented, aka less arrow-ness.